fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java

```diff
@@ -461,14 +474,28 @@ private void maybePerformScan() throws IOException {
                     } else if (exhausted) {
                         nextResult = RecordCursorResult.exhausted();
                     } else if (limitRemaining <= 0) {
-                        RecordCursorContinuation continuationFromDoc = LuceneCursorContinuation.fromScoreDoc(searchAfter, partitionId, partitionTimestamp);
+                        final Tuple primaryKey;
+                        if (nextResult != null && nextResult.get() != null) {
+                            primaryKey = nextResult.get().getPrimaryKey();
+                        } else {
+                            primaryKey = null;
+                        }
+                        RecordCursorContinuation continuationFromDoc = LuceneCursorContinuation.fromScoreDoc(
+                                searchAfter,
+                                partitionId,
+                                partitionKey,
+                                primaryKey);
                         nextResult = RecordCursorResult.withoutNextValue(continuationFromDoc, NoNextReason.RETURN_LIMIT_REACHED);
                     } else {
                         final Optional<NoNextReason> stoppedReason = limitManager.getStoppedReason();
                         if (stoppedReason.isEmpty()) {
                             throw new RecordCoreException("limit manager stopped LuceneRecordCursor but did not report a reason");
                         } else {
-                            nextResult = RecordCursorResult.withoutNextValue(LuceneCursorContinuation.fromScoreDoc(searchAfter, partitionId, partitionTimestamp), stoppedReason.get());
+                            nextResult = RecordCursorResult.withoutNextValue(LuceneCursorContinuation.fromScoreDoc(
+                                    searchAfter,
+                                    partitionId,
+                                    partitionKey,
+                                    null), stoppedReason.get());
```

Comment:
`null` seems incorrect here, we should get the primary key from the `searchAfter`.

Reply:
We won't need to pass it explicitly (will grab it from the `ScoreDoc`)